### PR TITLE
config lock timeout

### PR DESCRIPTION
**Pull request by weiwei04**
**What this PR does**

make SimpleStorage ResourceLock timeout configurable

**why we need it**

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by kdada** [comment]
@weiwei04 
You change the default lock behavior from `TimeoutImmediate` to `TimeoutInfinite`. It causes `SimpleStorage` won't return a `Locked` error. Could you describe your thought?

**Comment by weiwei04** [comment]
@kdada I don't have a strong reason to choose TimeoutInfinite over TimeoutImmediate, just because TimeoutInfinite is more like sync.Mutex :)

In our experience with helm-registry, we occasionally encounter `is locked` error in our dev environment, because our dev environment is running on small flavor vm instances, and we store registry filesystem on ceph rbd(which also running on vm), slower than local disk, 5 Millisecond timeout sometimes is not enough.

Just another thought, how about:
1. Set the default lock timeout behavior to TimeoutImmediate.
2. Provide a per request timeout, and use this per request timeout as a hit to set lock timeout, if user did'nt provide request timeout, the lock timeout behavior is TimeoutImmediate.

File: pkg/storage/simple/manager.go [hunk 1]
```diff
@@ -74,15 +75,26 @@ func (factory *simpleSpaceManagerFactory) Create(parameters map[string]interface
...
-		return nil, ErrorContentMissing.Format("storagedriver")
+		return nil, ErrorContentMissing.Format(common.ParameterNameStorageDriver)
 	}
 	storageDriver, err := driver.Create(fmt.Sprint(storageDriverName), parameters)
 	if err != nil {
 		return nil, ErrorInternalUnknown.Format(err)
 	}
-	return NewSpaceManager(storageDriver, locker, lock.TimeoutImmediate), nil
+
+	var lockTimeout = lock.TimeoutInfinite
```
Comment (kdada) [inline pkg/storage/simple/manager.go]: `lockTimeout` defaults to `lock.TimeoutImmediate`.  If a user want a longer timeout, he can set `locktimeout` in config.

**Comment by kdada** [comment]
Per request timeout will make a lot of changes. I prefer a default timeout with `lock.TimeoutImmediate` and user can change it by config.

**Comment by weiwei04** [comment]
Agree :)

**Pull request merged**